ssr/src/utils/token/mod.rs

```diff
@@ -281,6 +288,10 @@ pub async fn claim_tokens_from_first_neuron_if_required(
         return Ok(());
     }
 
+    if only_check {
+        // this prevents claiming during initial checking to prevent races
+        return Err(ServerFnError::ServerError("Claim required".to_string()));
+    }
```

Comment:
Create a separate function which returns `Result<Option<Neuron>>` that must be claimed, Then `claim_tokens_from_first_neuron` can be called directly in the new component